src/Admin/AbstractAdmin.php

```diff
@@ -1888,11 +1936,12 @@ public function getFilterFieldDescriptions()
 
     public function addChild(AdminInterface $child)
     {
-        for ($parentAdmin = $this; null !== $parentAdmin; $parentAdmin = $parentAdmin->getParent()) {
-            if ($parentAdmin->getCode() !== $child->getCode()) {
-                continue;
-            }
+        $parentAdmin = $this;
+        while ($parentAdmin->isChild() && $parentAdmin->getCode() !== $child->getCode()) {
```

Comment:
Does this code do the same thing? Looks like before it was comparing each parent with the child and this is comparing just the root parent.

Reply:
I still compare each parent with the $child.

In
`$parentAdmin->isChild() && $parentAdmin->getCode() !== $child->getCode()`
there is the check 
`$parentAdmin->getCode() !== $child->getCode()`